component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/service/MavenRepositoryDefaultResolver.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (C) 2006-2022 Talend Inc. - www.talend.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.runtime.manager.service;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.stream.Stream;
+
+import org.talend.sdk.component.runtime.manager.service.path.PathHandler;
+import org.talend.sdk.component.runtime.manager.service.path.PathHandlerImpl;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * m2 discovery process is used for plugins/connectors loading.
+ *
+ * The default discovery process is at it follows:
+ * <ul>
+ * <li>read {@code talend.component.manager.m2.repository} system property</li>
+ * <li>check if we're running in studio and uses its m2 if system property is not set to maven.repository=global</li>
+ * <li>parse maven's {@code settings.xml} (or if provided {@code talend.component.manager.m2.settings}) for checking if
+ * a property localRepository is defined</li>
+ * <li>checking for maven env properties (ie $M2_HOME)</li>
+ * <li>fallbacks to ${user.home}/.m2/repository</li>
+ * </ul>
+ */
+@Slf4j
+@NoArgsConstructor
+@Data
+public class MavenRepositoryDefaultResolver implements MavenRepositoryResolver {
```

Comment:
Nice,
Could even be better with several classes (System, Studio, Maven, Env) ... But good enough here :+1: